### DISCUSSION TITLE
Docker Extension for Airflow

### DISCUSSION THREAD

**Discussion by amyb12345** [apache/airflow#26693] [General]
### Description

Docker Extensions are a way to integrate additional functionality and tools into Docker Desktop.  Developers using Docker Desktop have been requesting that an extension be built for Airflow. Docker Extensions have an [open SDK](https://github.com/docker/extensions-sdk) and can be built by anyone. We'd love to see one created for Airflow to help improve developers workflows.  The Docker Extensions team is available to help support anyone interested in building this extension.  We can be reached via [email](mailto:extensions@docker.com), on [our community form](https://forums.docker.com/) or on our [SDK repo](https://github.com/docker/extensions-sdk).

### Use case/motivation

The goal of Docker Extensions is to help improve developers workflows. Developers are suggesting to Docker that an Airflow extension should exist. We'd love to see this happen to improve Airflow/Docker developers workflows.

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by potiuk** [comment]
Many of our users are actually migrating away from Docker Desktop - using Podman and others, because of the DockerDesktop licencing change. 

Apache Software Foundation has a very strong believe that software we develop should be free and it should not limit our users from doing anything by the licence. This is the core foundational value for Apache Software Foundation. 

If we were to support Docker Desktop, it's licence cannot really expect people to pay for the use of it under certain conditions. Our users should be free to do whatever they want and also free to redistribute the software with any modifications they see fit and their users should not be bound by any conditions that expect them to pay.

Is Docker team willing to change their licencing terms to comply with this expectation from the Apache Software Foundation? Then we might consider supporting Docker Desktop. 

@amyb12345 - can you please find an answer to that?

**Reply by amyb12345** [reply]
@potiuk Unfortunately, the extensions team is not able to change the licensing terms of Docker Desktop. Docker Desktop does have a [free tier](https://www.docker.com/products/personal/) available non-commercial open source developers. I understand that this may still not meet the expectations of the Apache Software Foundation and you may not be able to support this request from developers who are using both Docker Desktop and Airflow.  Extensions are available to all Docker Desktop users for free if that's helpful to know.

**Reply by eladkal** [reply]
> Docker Desktop does have a free tier available non-commercial open source developers

We developers still work somewhere and I dont think many people have seperated computer just for open source :)
So this is not really practical.

I moved away from Docker Deaktop due to this.

**Reply by potiuk** [reply]
Hey @amyb12345  - you probably did not understand what I was explaining. Similarly like Lightbend CEO in https://issues.apache.org/jira/browse/LEGAL-619. - this is not the matter of free tier or even an exclusion for the ASF project or all open-source project. The whole reason for existence of Apache Software Foundation is that we give software for free and we PROMISE to our users that whatever software we release will not bind them with hidden licence rules and obligation to pay.

We are absolutely welcoming making money on the Open-Source software. Many of our users do. Mostly by derivative work, services and  similar. But we cannot ONE of our users impose licence limits on ASF released software. This is something we promised our users and all ASF projects will have exactly the same limitation. It's just basic, foundational non-negotiable value that all the ASF projects have.
